homeassistant/components/mqtt/siren.py

```diff
@@ -0,0 +1,341 @@
+"""Support for MQTT sirens."""
+from __future__ import annotations
+
+import functools
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import siren
+from homeassistant.components.siren import SirenEntity
+from homeassistant.components.siren.const import (
+    ATTR_AVAILABLE_TONES,
+    ATTR_DURATION,
+    ATTR_TONE,
+    ATTR_VOLUME_LEVEL,
+    SUPPORT_DURATION,
+    SUPPORT_TONES,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_SET,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_OPTIMISTIC,
+    CONF_PAYLOAD_OFF,
+    CONF_PAYLOAD_ON,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import PLATFORMS, MqttCommandTemplate, MqttValueTemplate, subscription
+from .. import mqtt
+from .const import (
+    CONF_COMMAND_TEMPLATE,
+    CONF_COMMAND_TOPIC,
+    CONF_ENCODING,
+    CONF_QOS,
+    CONF_RETAIN,
+    CONF_STATE_TOPIC,
+    DOMAIN,
+)
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, MqttEntity, async_setup_entry_helper
+
+DEFAULT_NAME = "MQTT Siren"
+DEFAULT_PAYLOAD_ON = "ON"
+DEFAULT_PAYLOAD_OFF = "OFF"
+DEFAULT_OPTIMISTIC = False
+
+ENTITY_ID_FORMAT = siren.DOMAIN + ".{}"
+
+CONF_AVAILABLE_TONES = "available_tones"
+CONF_COMMAND_OFF_TEMPLATE = "command_off_template"
+CONF_DURATION_COMMAND_TEMPLATE = "duration_command_template"
+CONF_DURATION_COMMAND_TOPIC = "duration_command_topic"
+CONF_STATE_ON = "state_on"
+CONF_STATE_OFF = "state_off"
+CONF_SUPPORT_DURATION = "supported_duration"
+CONF_SUPPORT_TURN_OFF = "supported_turn_off"
+CONF_SUPPORT_TURN_ON = "supported_turn_on"
+CONF_SUPPORT_VOLUME_SET = "supported_volume_set"
```

Comment:
Should it be like this instead:
```suggestion
CONF_SUPPORT_DURATION = "support_duration"
CONF_SUPPORT_TURN_OFF = "support_turn_off"
CONF_SUPPORT_TURN_ON = "support_turn_on"
CONF_SUPPORT_VOLUME_SET = "support_volume_set"
```

Also, why are these needed, isn't it enough to check if the command templates are set?

Reply:
Did some rework